lib/model/game_state.dart

```diff
@@ -14,23 +14,23 @@ import 'hand.dart';
 class GameState with ChangeNotifier {
   final Randomiser randomiser;
 
-  Card currentCard;
+  Card? currentCard;
   final List<Card> deck;
   final List<Card> discard = [];
   final List<Card> exile = [];
   final List<Card> scrying = [];
 
   final Hand playerHand = Hand();
 
-  GameResult result;
+  GameResult? result;
 
-  GameState(this.randomiser, this.deck, [List<Resource> initialResources]) {
-    playerHand.addResources(initialResources);
+  GameState(this.randomiser, this.deck, [List<Resource>? initialResources]) {
+    playerHand.addResources(initialResources!);
```

Comment:
Could we drop the `!` and instead make the `initialResources` not nullable? I guess we can default it to an empty list instead if we want the argument to remain optional.